src/ast/analysis/typesystem/TypeConstrainsAnalysis.cpp

```diff
@@ -174,6 +174,10 @@ void TypeConstraintsAnalysis::visit_(type_identity<Counter>, const Counter& coun
     addConstraint(isSubtypeOf(getVar(counter), typeEnv.getConstantType(TypeAttribute::Signed)));
 }
 
+void TypeConstraintsAnalysis::visit_(type_identity<IterationCounter>, const IterationCounter& counter) {
+    addConstraint(isSubtypeOf(getVar(counter), typeEnv.getConstantType(TypeAttribute::Unsigned)));
```

Comment:
I am wondering if a `Signed` would be more appropriate, my guess is that `number`s are more frequent than `unsigned` so it would save some `as` conversions. What do you think?

Reply:
Personally I'd prefer keeping the unsigned because it makes more sense for a loop counter.
The way I envision people will use it is by passing it directly to a user defined functor `@my_functor(recursive_loop_cnt(), x, y)` so the functor behaves differently dependending on the loop iteration.
But I don't expect people to use it much in operations involving other numbers.